src/machi_cr_client.erl

```diff
@@ -434,6 +437,9 @@ do_append_midtail2([FLU|RestFLUs]=FLUs, Prefix, File, Offset, Chunk,
             Resume = {append, Offset, iolist_size(Chunk), File},
             do_repair_chunk(FLUs, Resume, Chunk, [], File, Offset,
                             iolist_size(Chunk), Depth, STime, S);
+        {error, trimmed} = Err ->
+            %% TODO: nothing can be done
+            {reply, Err, S};
```

Comment:
This one is interesting.  If we got here, then the head (at least) was not trimmed.  To find a trimmed block here is unusual.  However ... it's more possible if this FLU comes from this epoch's repairing server list.

In AP mode, this would be a signal that all replicas should be trimmed; a repair procedure is probably in progress right now, but we (the append client) and the repair have had a race, and we won.  If repair is indeed running, it will figure out sometime that other FLUs also need to have this byte range trimmed.

In CP mode, the repair process code isn't finished yet, alas.  But again, this case could easily happen if this FLU is in the repairing server list.  Returning an error here, previous head & middle/tail writes may have succeeded, which makes this append only partially successful ... and leaving the client to retry if it wants to.  That seems ok with me.  The alternative is to ignore this error when in CP mode (because this trimmed region is itself the result of a partially-successful trim and will be *undone* by repair) and continue with the chain.  But that's extra complexity that I don't believe that we need.